documentation/projects/proposals/rekognition_data/20240530-implementation_plan_augment_catalog_with_rekognition_tags.md

```diff
@@ -0,0 +1,669 @@
+specifically[^4][^5][^6].
+
+Certain demographic axes seem the most likely to result in an incorrect or
+insensitive label (e.g. gender assumption of an individual in a photo). For the
+reasons described in the above cited works, we should **exclude** labels that
+have a demographic context in the following categories:
+
+- Age
+- Gender
+- Sexual orientation
+- Nationality
+- Race
+
+There are other categories which might be useful for search relevancy and are
+less likely to be applied in an insensitive manner. These labels **should not**
+be excluded. Some examples include:
+
+- Occupation
+- Marital status
+- Health and disability status
+- Political affiliation or preference
+- Religious affiliation or preference
+
+### Accuracy selection
+
+[^removal]: Note that this step will be going away as part of #430
+
+We already filter out existing tags from the catalog when copying data into the
+API database during the data refresh's
+[cleanup step](https://github.com/WordPress/openverse/blob/3747f9aa40ed03899becb98ecae2abf926c8875f/ingestion_server/ingestion_server/cleanup.py#L119-L150)[^removal].
+The minimum accuracy value used for this step is
+[0.9 (or 90%)](https://github.com/WordPress/openverse/blob/3747f9aa40ed03899becb98ecae2abf926c8875f/ingestion_server/ingestion_server/cleanup.py#L57-L56)
+. AWS's own advice on what value to use is that
+[it depends entirely on the use case of the application](https://aws.amazon.com/rekognition/faqs/#Label_Detection).
+
+I took a small sample of the labels we have available (~100MB out of the 196GB
+dataset, about 45k images with labels) and performed some exploratory analysis
+on the data. I found the following pieces of information:
+
+- **Total images**: 45,059
+- **Total labels**: 555,718
+- **Average confidence across all labels**: 79.927835
+- **Median confidence across all labels**: 81.379463
+- **Average confidence per image**: 81.073921
+- **Median confidence per image**: 82.564148
+- **Number of labels with confidence higher than 90**: 210,341
+- **Percentage of labels with confidence higher than 90**: 37.85031%
+- **Average number of labels per image higher than 90**: 4.6629
+
+_For a full explanation on this exploration, see:
+[Analysis explanation](#analysis-explanation)_
+
+Based on the number of labels we would still be receiving with a confidence
+higher than 90, and that 0.9 is already our existing minimum standard, **we
+should retain 0.9 or 90% as our minimum label accuracy value** for inclusion in
+the catalog.
+
+This necessarily means that we will not be including a projected 62% of the
+labels which are available in the Rekognition dataset. Accuracy, as it directly
+relates to search relevancy, is more desirable here than completeness. We will
+retain the original Rekognition source data after ingesting the high-accuracy
+tags, and so if we decide to allow a lower accuracy threshold, we can always
+re-add the lower confidence values at a later time.
```

Comment:
+1! Would it be possible to get a sense of what percentage of labels will be included, once you remove labels with _any_ demographic assumptions from human subjects? Curious if that significantly lowers this percentage even further -- although I agree with your reasoning here in any case!

Reply:
Given a cursory glance over the list, I think the labels associated with the most results would be for things like "man" and "woman". You can get some idea of that from [the analysis Zack did](https://gist.github.com/zackkrida/ecba17d8fc89d8e4d8aa78e5e19199c0). In terms of percentage of _total labels_, it's probably quite small! Maybe dozens out of the thousands available.

Reply:
Female, Girl, Woman, Teen, Man, Kid, Child, Boy, Baby, Priest, Senior Citizen, Tribe, Lady, and Monk are all of the potential demographic terms I see in our list of data.